### PR TITLE
Add SupportedOSPlatform to Hosting.WindowsService HostBuilder extensions

### LINKED ISSUE #50013: UseWindowsService does not generate warning CA1416 "...is supported on 'windows'" in a project targeting net5.0
I have a test project targeting net5.0. 

* When I use IWebHost.RunAsService() compiler generates warning  CA1416 'WebHostWindowsServiceExtensions.RunAsService(IWebHost)' is supported on 'windows'.
* When I use IHostBuilder.UseWindowsService() compiler does not generate CA1416 warning.

### PR DESCRIPTION
**Pull request by hwoodiwiss**
fixes dotnet/runtime#50013

Includes System.Runtime.Versioning attributes into the build and applies them to the HostBuilder extensions.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=53941) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=53941) before we can accept your contribution.<br/><br/>:x: hwoodiwiss [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=53941)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=53941) it.</sub>

**Comment by hwoodiwiss** [comment]
Feedback is appreciated, I was on the fence between applying the attribute to the extension methods, and just applying it as an assembly attribute.

**Comment by eerhardt** [comment]
> I was on the fence between applying the attribute to the extension methods, and just applying it as an assembly attribute.

I think it makes the most sense to be assembly wide. The whole assembly is "WindowsServices". Those are only ever going to be supported on Windows.

We do it assembly wide elsewhere:

https://github.com/dotnet/runtime/blob/197cfb60f37626ce9e07285517b129fc01f9621e/src/libraries/Microsoft.Win32.Registry/Directory.Build.props#L6
https://github.com/dotnet/runtime/blob/197cfb60f37626ce9e07285517b129fc01f9621e/src/libraries/System.Security.Principal.Windows/Directory.Build.props#L6

cc @jeffhandley @buyaa-n

**Comment by hwoodiwiss** [comment]
Great feedback and great point. I've update to set the attribute assembly wide as suggested.

File: src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceLifetimeHostBuilderExtensions.cs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 using System;
 using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceLifetimeHostBuilderExtensions.cs:6]: Is this change still necessary?

File: src/libraries/Microsoft.Extensions.Hosting.WindowsServices/ref/Microsoft.Extensions.Hosting.WindowsServices.cs [hunk 2]
```diff
@@ -8,8 +8,8 @@ namespace Microsoft.Extensions.Hosting
 {
     public static partial class WindowsServiceLifetimeHostBuilderExtensions
     {
-        public static Microsoft.Extensions.Hosting.IHostBuilder UseWindowsService(this Microsoft.Extensions.Hosting.IHostBuilder hostBuilder) { throw null; }
-        public static Microsoft.Extensions.Hosting.IHostBuilder UseWindowsService(this Microsoft.Extensions.Hosting.IHostBuilder hostBuilder, System.Action<Microsoft.Extensions.Hosting.WindowsServiceLifetimeOptions> configure) { throw null; }
+       public static Microsoft.Extensions.Hosting.IHostBuilder UseWindowsService(this Microsoft.Extensions.Hosting.IHostBuilder hostBuilder) { throw null; }
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Hosting.WindowsServices/ref/Microsoft.Extensions.Hosting.WindowsServices.cs:11]: I believe this change should be reverted.

File: src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceLifetimeHostBuilderExtensions.cs (hunk 1 above)
Comment (hwoodiwiss) [reply src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceLifetimeHostBuilderExtensions.cs]: It was not, good catch, sorry about that.

File: src/libraries/Microsoft.Extensions.Hosting.WindowsServices/ref/Microsoft.Extensions.Hosting.WindowsServices.cs (hunk 2 above)
Comment (hwoodiwiss) [reply src/libraries/Microsoft.Extensions.Hosting.WindowsServices/ref/Microsoft.Extensions.Hosting.WindowsServices.cs]: This has now been reverted.

**Review by eerhardt — APPROVED** [review]
Looks great. Thanks for the contribution!
